### PR TITLE
Reuse source buffers

### PR DESCRIPTION
**Pull request by manthey**
This is a further extension of the `allocate-less-memory` branch, but the changes introduced in the last commit are independent of that, so I'm adding this as a separate PR.  If this is reviewed directly, both can be satisfied with this PR.

Rather than using the DataBuffers structure, allocate source buffers directly, reusing them if they haven't changed.  Also, for points and lines, we require the vertex indices to exist, but we don't do anything with them, so create but don't fill those arrays.

This shouldn't speed up the first load of data much, but when data is reloaded with the same number of points or lines, it improves the speed.  Compared to allocating the buffers every time, this reduces the time to load 1,000,000 points from 0.8 s to 0.7 s and for 1,000,000 lines from 4.0 s to 3.5 s (roughly 12 % improvement).

### CODE REVIEW

File: testing/test-cases/selenium-tests/glLinesSpeed/testGlLinesSpeed.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        self.loadTimeTest(testName)
+        self.framerateTest(testName)
+
+    def loadTimeTest(self, testName, revision=None):
+        # Threshold is in milliseconds.  We need the value to be less than the
+        # threshold.
+        el = self.getElement('#loadResults')
+        value = float(el.get_attribute('results'))
+        threshold = float(os.environ.get('LOAD_SPEED_THRESHOLD', '1000'))
+        print 'Average load time %1.0f ms (must be at least %1.0f ms)' % (
```
Comment (jbeezley) [inline testing/test-cases/selenium-tests/glLinesSpeed/testGlLinesSpeed.py]: s/at least/less than/

File: testing/test-cases/selenium-tests/glLinesSpeed/testGlLinesSpeed.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        if value > threshold or math.isnan(value):
+            raise ThresholdException({'value': value, 'threshold': threshold})
+
+    def framerateTest(self, testName, revision=None):
+        # Threshold is in frames-per-second.  We need the value to be greater
+        # than the threshold.
+        el = self.getElement('#framerateResults')
+        value = float(el.get_attribute('results'))  # in milliseconds
+        threshold = float(os.environ.get('FRAMERATE_THRESHOLD', '10'))
+        print 'Average framerate %4.2f fps (must be less than %4.2f fps)' % (
```
Comment (jbeezley) [inline testing/test-cases/selenium-tests/glLinesSpeed/testGlLinesSpeed.py]: s/less than/at least/

**Comment by jbeezley** [comment]
The test look great.  I'll leave the review of the GL to @aashish24.

**Comment by aashish24** [comment]
Other than that one suggestion, looks great to me! I am merging it :+1:

**Pull request merged**